nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -863,7 +863,13 @@ in
                   The address must be in the default VLAN (10.0.2.0/24).
               '';
           }
-        ]));
+        ])) ++ [
+          { assertion = pkgs.stdenv.hostPlatform.is32bit && cfg.memorySize < 2047;
```

Comment:
I think this should be `pkgs.stdenv.hostPlatform.is32bit -> cfg.memorySize < 2047`

Reply:
Argh, right.

Reply:
Fix in https://github.com/NixOS/nixpkgs/pull/234455.